velox/expression/Expr.cpp

```diff
@@ -184,11 +196,15 @@ void Expr::computeMetadata() {
     propagatesNulls_ = vectorFunction_->isDefaultNullBehavior();
     deterministic_ = vectorFunction_->isDeterministic();
   }
+
+  std::set<FieldReference*> allFields;
   for (auto& input : inputs_) {
     input->computeMetadata();
     deterministic_ &= input->deterministic_;
     propagatesNulls_ &= input->propagatesNulls_;
     mergeFields(distinctFields_, input->distinctFields_);
+    // find the fields referenced by multiple inputs
+    findMultiRefFields(allFields, multiRefFields_, input->distinctFields_);
```

Comment:
it seems that the new logic can be folded into mergeFields since it already contains de-deplication logic. This way we won't need to maintain allFields as we can use distinctFields_ instead.